src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -169,73 +171,62 @@ public void reset() {
    */
   static Supplier<SelectionComponent<JComponent>> intValueRange(final ClientSetting clientSetting, final int lo,
       final int hi) {
+    return intValueRange(clientSetting, lo, hi, false);
+  }
+
+  /**
+   * Text field that only accepts numbers between a certain range.
+   */
+  static Supplier<SelectionComponent<JComponent>> intValueRange(final ClientSetting clientSetting, final int lo,
+      final int hi, final boolean allowUnset) {
     return () -> new SelectionComponent<JComponent>() {
-      String value = clientSetting.value();
-      final JTextField component = new JTextField(value, String.valueOf(hi).length());
+      private final JSpinner component = new JSpinner(new SpinnerNumberModel(
+          toValidIntValue(clientSetting.value()), lo - (allowUnset ? 1 : 0), hi, 1));
 
       @Override
       public JComponent getUiComponent() {
-        component.setToolTipText(validValueDescription());
+        return component;
+      }
 
-        SwingComponents.addTextFieldFocusLostListener(component, () -> {
-          if (isValid()) {
-            clearError();
-          } else {
-            indicateError();
-          }
-        });
+      private int toValidIntValue(final String value) {
+        return value.isEmpty() && allowUnset ? lo - 1 : Integer.parseInt(value);
+      }
 
-        return component;
+      private String toValidStringValue(final int value) {
+        return allowUnset && value == lo - 1 ? "" : String.valueOf(value);
       }
 
       @Override
       public boolean isValid() {
-        final String value = component.getText();
-
-        if (value.trim().isEmpty()) {
-          return true;
-        }
-
-        try {
-          final int intValue = Integer.parseInt(value);
-          return intValue >= lo && intValue <= hi;
-        } catch (final NumberFormatException e) {
-          return false;
-        }
+        return true;
       }
 
       @Override
       public String validValueDescription() {
-        return "Number between " + lo + " and " + hi;
+        return "";
```

Comment:
Is it the case that the value is always valid?

Reply:
Yes, because the user can't enter non numerical values into the spinner.

Reply:
EDIT: The user can enter invalid values into the spinner, but the spinner doesn't store those invalid values. Whenever a call to getValue is made, the last valid value is being returned